File: test/fixtures/sample.java
class App {
    String greet(String name) {
        return "Hello, " + name;
    }

    boolean validate(String email) {
        if (email.length() > 0) {
            return true;
        }
        return false;
    }

    void processUser(String name, String email) {
        String greeting = greet(name);
        System.out.println(greeting);
        if (validate(email)) {
            saveUser(name, email);
        }
    }

    void saveUser(String name, String email) {
        System.out.println("Saving user: " + name + ", " + email);
    }

    public static void main(String[] args) {
        App app = new App();
        app.processUser("Alice", "alice@test.com");
    }
}

File: test/fixtures/sample.rs
fn greet(name: &str) -> String {
    format!("Hello, {}", name)
}

fn validate(email: &str) -> bool {
    if email.len() > 0 {
        return true;
    }
    false
}

fn process_user(name: &str, email: &str) {
    let greeting = greet(name);
    println!("{}", greeting);
    if validate(email) {
        save_user(name, email);
    }
}

fn save_user(name: &str, email: &str) {
    println!("Saving user: {}, {}", name, email);
}

fn main() {
    process_user("Alice", "alice@test.com");
}
